Content.Server/GameTicking/GameTicker.JobController.cs

```diff
@@ -25,131 +25,80 @@ public partial class GameTicker
         [ViewVariables]
         private readonly Dictionary<string, int> _spawnedPositions = new();
 
-        private Dictionary<IPlayerSession, string> AssignJobs(List<IPlayerSession> available,
+        private Dictionary<IPlayerSession, (string, StationId)> AssignJobs(List<IPlayerSession> available,
             Dictionary<NetUserId, HumanoidCharacterProfile> profiles)
         {
-            // Calculate positions available round-start for each job.
-            var availablePositions = GetBasePositions(true);
-
-            // Output dictionary of assigned jobs.
-            var assigned = new Dictionary<IPlayerSession, string>();
-
-            // Go over each priority level top to bottom.
-            for (var i = JobPriority.High; i > JobPriority.Never; i--)
+            var assigned = new Dictionary<IPlayerSession, (string, StationId)>();
+            // Current strategy is to fill each station one by one.
+            foreach (var (id, station) in _stationSystem.StationInfo)
             {
-                void ProcessJobs(bool heads)
+                // Get the ROUND-START job list.
+                var availablePositions = station.MapPrototype.AvailableJobs.ToDictionary(x => x.Key, x => x.Value[0]);
+                for (var i = JobPriority.High; i > JobPriority.Never; i--)
                 {
-                    // Get all candidates for this priority & heads combo.
-                    // That is all people with at LEAST one job at this priority & heads level,
-                    // and the jobs they have selected here.
-                    var candidates = available
-                        .Select(player =>
-                        {
-                            var profile = profiles[player.UserId];
+                    void ProcessJobs(bool heads)
+                    {
+                        var candidates = available.Select(player =>
+                            {
+                                var profile = profiles[player.UserId];
 
-                            var availableJobs = profile.JobPriorities
-                                .Where(j =>
-                                {
-                                    var (jobId, priority) = j;
-                                    if (!_prototypeManager.TryIndex(jobId, out JobPrototype? job))
+                                var availableJobs = profile.JobPriorities
+                                    .Where(j =>
```

Comment:
It was already largely like this before, but I have to say that I really struggle to follow the logic in this function when reading it. Probably mainly due to the double-nested for loop with loop-local functions and chunky lambda functions.

If other's don't think its that important it's probably fine to leave as it is (assuming it's working), especially given that this PR is apparently blocking other job/station related PRs.

But I'd still prefer to not get a headache if I ever have to go read job code. Without just reorganizing the whole logic I think it would already be made easier to comprehend if `ProcessJobs` were just moved out of the double for loops (i.e., just local to AssignJobs). Also the `var candidates = ....` lambda function query could probably be turned into local-function (`GetCandidates`) or something? That's basically what what I did locally when trying to work through it.